src/main/java/com/example/demo/tournament/CSVcontroller.java

```diff
@@ -0,0 +1,51 @@
+package com.example.demo.tournament;
+
+import com.example.demo.model.Team;
+import com.opencsv.CSVReader;
+import com.opencsv.CSVReaderBuilder;
+import com.opencsv.CSVWriter;
+import com.opencsv.bean.StatefulBeanToCsv;
+import com.opencsv.bean.StatefulBeanToCsvBuilder;
+import com.opencsv.exceptions.CsvDataTypeMismatchException;
+import com.opencsv.exceptions.CsvRequiredFieldEmptyException;
+import org.springframework.stereotype.Controller;
+
+import java.io.*;
+import java.util.List;
+
+@Controller
+public class CSVcontroller {
+
+    public static void Write(List<Team> listTeams) {
```

Comment:
Don't use static in Spring components. This gives you an option to use this class without spring (you don't need to inject, you can call directly) which makes this dependency in other components less visible and more highly coupled

Reply:
Then i will need to create an application context and get the CSVcontroller from there wherever i want to use it?